lib/streamlit/runtime/scriptrunner_utils/script_run_context.py

```diff
@@ -116,7 +143,9 @@ def reset(
         self.widget_user_keys_this_run = set()
         self.form_ids_this_run = set()
         self.query_string = query_string
-        self.pages_manager.set_current_page_script_hash(page_script_hash)
+        self._page_script_hash = page_script_hash
+        if self.pages_manager.mpa_version == 2:
+            self._active_script_hash = self.pages_manager.main_script_hash
```

Comment:
nit: instead of doing this check and the check in the property [active_script_hash](https://github.com/streamlit/streamlit/pull/9441/files#diff-ed869edbb0923d306ba5e7efaafa8081df7f1a3e8372842545e9b98223d3d496R116-R119) here, could we just have a method `get_active_script_hash` in the `pages_manager` for the two strategies and change the property above and these lines here:

```python
@property
def active_script_hash(self):
  return self._active_script_hash

self._page_script_hash = page_script_hash
self._active_script_hash = self.pages_manager.get_active_script_hash()
```